server/sonar-server/src/main/java/org/sonar/server/issue/ws/AssignAction.java

```diff
@@ -121,8 +120,9 @@ private void assign(String issueKey, @Nullable String assignee) {
       }
       IssueChangeContext context = IssueChangeContext.createUser(new Date(system2.now()), userSession.getLogin());
       if (issueFieldsSetter.assign(issue, user, context)) {
-        issueUpdater.saveIssue(dbSession, issue, context, null);
+        return issueUpdater.saveIssueAndPreloadSearchResponseData(dbSession, issue, context, null);
```

Comment:
Note that the user could also be part of the preloaded response. It implies to change IssueChangeContext by replacing user login by UserDto.
This improvement does not sound to be a must-have.